### ISSUE TITLE
functions with the same name as their return type are misparsed

### ISSUE THREAD

**Issue by topisani** [foonathan/cppast#29]
* cppast version: latest master
* clang version: tested with 5.0 and latest svn

If a function has the same name as its return type, it is misparsed as a class declaration it seems.

```cpp
struct name {};
name name();
```
Now, this is a weird thing to do, but this is not as weird:
```cpp
std::string string();
```
In fact it came up while implementing the filesystem ts, because of `filesystem::path::string()`.

This is what standardese reports:
```
[libclang parser] [error] test.cpp:4: expected a bracket, got 'name'
```

**Comment by foonathan** [comment]
This is again a bug on my parsing code, I thought I had workarounds for it.

**Comment by foonathan** [comment]
I knew I fixed it, the fix was just in a different branch I hadn't merged due to performance problems.

This should be fixed by 98e709fdfcfdaab6dcbc4447cb7cdce77bc2cd6c.
